dive_core/command_hierarchy.cpp

```diff
@@ -1801,13 +1803,28 @@ void CommandHierarchyCreator::AppendRegNodes(const IMemoryManager &mem_manager,
         if (reg_info_ptr == nullptr)
             reg_info_ptr = &temp;
 
-        DIVE_ASSERT(!reg_info_ptr->m_is_64_bit);
+        uint64_t reg_value = reg_pair.m_reg_value;
+        if (reg_info_ptr->m_is_64_bit)
+        {
+            RegPair  new_reg_pair;
+            uint64_t new_pair_addr = va_addr + dword * sizeof(uint32_t);
+            DIVE_VERIFY(mem_manager.CopyMemory(&new_reg_pair,
+                                               submit_index,
+                                               new_pair_addr,
+                                               sizeof(new_reg_pair)));
+
+            // Sometimes the upper 64-bits are not set
+            // Probably because they're 0s and there's no need to set it
+            if (new_reg_pair.m_reg_offset == reg_pair.m_reg_offset + 1)
+            {
+                dword += 2;
```

Comment:
shouldn't this `dword +=2` be outside of the `if` so that for 64bit register, even the upper 32bit is not used, we should still advance dword here?

Reply:
There's already a dword+=2 earlier. If the offsets are consecutive, then it does the dword+=2 here again, which ends up doing the equivalent of dword+=4 for this iteration of the loop.
However, if the offsets are not consecutive, then we only process the first 2 32-bit dwords during this iteration of the loop. Then we go back to the beginning of the loop and process the next 2 32-bit dwords as though it was a whole separate register (which it is).

Reply:
oh, so even the reg_info_ptr->m_is_64_bit flag is true, if `offset1 != offset0 + 1`, the register will still only take 2 dwords instead of 4, right?

Reply:
Yup

Reply:
Note: cffdump also spits out a warning in this particular case:

`
      if (regbase != r->regbase + 1) {
         printf("WARNING: 64b discontinuity (%x, expected %x)\n", regbase, r->regbase + 1);
         r->has_dword_lo = false;
         return true;
      }
`

And we indeed get this warning in the cffdump of the capture I found the problem in

Reply:
ah, ok. make sense. thanks!